scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -391,6 +393,19 @@ class ScioContext private[scio] (val options: PipelineOptions,
         context.optionsAs[ApplicationNameOptions].getAppName,
         state.toString,
         getBeamMetrics)
+
+    override def getJobTimeout: Duration = {
+      Option(optionsAs[ScioOptions].getJobTimeout) match {
+        case Some(duration) => try {
+          Duration.apply(duration)
+        } catch {
+          case e: NumberFormatException =>
```

Comment:
So this will throw after job was already submitted, right? I wonder if it would be better to do this check before we submit the job?

Reply:
My only concern there was that there would be some duplicated logic - since the timeout is string-serialized in `ScioOptions`, if we added it as a pre-check when `ScioContext` was instantiated, there's not really a place to store the resulting `Duration` object unless you want to introduce some settable `var`. So we'd be deserializing a `Duration` again after submitting the job.

Ultimately not a huge deal since it's still only two invocations :)

Reply:
actually, I think if I just declare it once as a private val anytime parseArguments has been called it should work

Reply:
Yeah, I think we can take that cost (two invocations), then have this weird situation where the job is submitted and the driver fails with a silly typo. @regadas what do you think?

Reply:
@ravwojdyla yeah I think we should try doing an early validation of the arg.